trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/credentials/http/HttpCredentialsProviderConfig.java

```diff
@@ -17,10 +17,15 @@
 import jakarta.validation.constraints.NotNull;
 
 import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
 public class HttpCredentialsProviderConfig
 {
     private URI endpoint;
+    private Optional<Map<String, String>> httpHeaders = Optional.ofNullable(null);
```

Comment:
This is the same as `Optional.empty()`

This could instead just be a map which defaults to being empty right?

Reply:
Can we make the default an empty map? The logic works the same way whether we have an empty map or `Optional.empty()` so I don't think we need to wrap this map in an optional